frontstage/views/surveys.py

```diff
@@ -1,37 +1,164 @@
+                                 reverse=True)
+
+    # Checks if new survey is added and renders "survey added" notification
+    if request.args.get('new_survey') and ['NEW_SURVEY_NOTIF_HIGHLIGHTING'] == 1:
+        logger.info('New survey added highlighted')
+        template_data = {"survey": {"justAdded": "true"}}
+        return render_template('surveys/surveys-todo.html', _theme='default',
+                               data=template_data, sorted_surveys_list=sorted_surveys_list,
+                               just_added_case_id=request.args.get('case_id'))
+
+    return render_template('surveys/surveys-todo.html', data=template_data,
+                           _theme='default', sorted_surveys_list=sorted_surveys_list)
 
 
 @surveys_bp.route('/history', methods=['GET'])
 @jwt_authorization(request)
 def surveys_history(session):
     party_id = session['party_id']
     surveys_list = get_surveys_list(party_id, 'history')
-    return render_template('surveys/surveys-history.html',  _theme='default',
+    template_data = {"survey": {"justAdded": "false"}}
+    return render_template('surveys/surveys-history.html',  _theme='default', data=template_data,
                            surveys_list=surveys_list, history=True)
 
 
+@surveys_bp.route('/add-survey', methods=['GET', 'POST'])
+@jwt_authorization(request)
+def add_survey(session):
+    form = EnrolmentCodeForm(request.form)
+
+    if request.method == 'POST' and form.validate():
+        logger.info('Enrolment code submitted')
+        enrolment_code = request.form.get('enrolment_code').lower()
+        request_data = {
+            'enrolment_code': enrolment_code,
+            'initial': True
+        }
+        response = api_call('POST', app.config['VALIDATE_ENROLMENT'], json=request_data)
+
+        # Handle API errors
+        if response.status_code == 404:
+            logger.info('Enrolment code not found')
+            template_data = {"error": {"type": "failed"}}
+            return render_template('surveys/surveys-add.html', _theme='default',
+                                   form=form, data=template_data), 202
+        elif response.status_code == 401 and not json.loads(response.text).get('active'):
+            logger.info('Enrolment code not active')
+            template_data = {"error": {"type": "failed"}}
+            return render_template('surveys/surveys-add.html', _theme='default',
+                                   form=form, data=template_data), 200
+        elif response.status_code == 400:
+            logger.info('Enrolment code already used')
+            template_data = {"error": {"type": "failed"}}
+            return render_template('surveys/surveys-add.html', _theme='default',
+                                   form=form, data=template_data), 200
+        elif response.status_code != 200:
+            logger.error('Failed to submit enrolment code')
```

Comment:
Can we add status=response.status_code to log line?

Reply:
This is logged in `ApiError`